pina/model/network.py

```diff
@@ -0,0 +1,95 @@
+        corresponding to the input components of the model.
+    :param list(str) output_variables: the list containing the labels
+        corresponding to the components of the output computed by the model.
+    :param iterable(torch.nn.Module) extra_features: the additional input
+        features to use as augmented input.
+
+    :Example:
+            >>> class SimpleNet(nn.Module):
+            >>>    def __init__(self):
+            >>>        super().__init__()
+            >>>        self.layers = nn.Sequential(
+            >>>        nn.Linear(3, 20),
+            >>>        nn.Tanh(),
+            >>>        nn.Linear(20, 1)
+            >>>        )
+            >>>    def forward(self, x):
+            >>>        return self.layers(x)
+            >>> net = SimpleNet()
+            >>> input_variables = ['x', 'y']
+            >>> output_variables =['u']
+            >>> model_feat = Network(net, input_variables, output_variables)
+            Network(
+                (extra_features): Sequential()
+                (model): Sequential(
+                    (0): Linear(in_features=2, out_features=20, bias=True)
+                    (1): Tanh()
+                    (2): Linear(in_features=20, out_features=1, bias=True)
+                )
+            )
+    """
+
+    def __init__(self, model, input_variables,
+                 output_variables, extra_features=None):
+        super().__init__()
+
+        if extra_features is None:
+            extra_features = []
+
+        self._extra_features = nn.Sequential(*extra_features)
+        self._model = model
+        self._input_variables = input_variables
+        self._output_variables = output_variables
+
+        try:
+            tmp = torch.rand((10, len(input_variables)))
+            self._model(tmp)
+        except:
+            raise ValueError('Error in constructing the PINA network.'
+                             ' Check compatibility of input/output'
+                             ' variables shape with the torch model'
+                             ' or check the correctness of the torch'
+                             ' model itself.')
+
+    def forward(self, x):
+        """Forward method for Network class
+
+        :param torch.tensor x: input of the network
+        :return torch.tensor: output of the network
+        """
+
+        x = x.extract(self._input_variables)
+
+        for feature in self._extra_features:
+            x = x.append(feature(x))
+
+        output = self._model(x).as_subclass(LabelTensor)
+        output.labels = self._output_variables
+
+        return output
+
+    @property
+    def input_variables(self):
+        return self._input_variables
+
+    @property
+    def output_variables(self):
+        return self._output_variables
+
+    @property
+    def extra_features(self):
+        return self._extra_features
+
+    @property
+    def model(self):
+        return self._model
```

Comment:
Properties without explicit setters are meant for denying the user to change such attributes. In this case, I don't see any critical issues even changing the attributes after object instantiation, why do you have "lock" them?

Reply:
Consider that the check of model consistency with input variables is done **only** when `Network` object is constructed. If one changes the model after the construction it could happen that the new model doesn't satisfy input/ouput variables length. This is just one example of possible failure. In practice if we want to avoid this we could put a check into the forward pass, but this would cause a loose in efficiency. Instead, by marking property the variables we ensure that (if someone wants to change model or input/output variable), it must construct another object. 

For me it makes sense, since I think Network has to be seen as a translator from a `torch.nn.Module` model to a PINA model, different torch models lead to different PINA model

Reply:
If you want I can remove it, but I believe marking as property we force the user to initialize a new PINA model everytime, avoiding runtime error in the forward pass, which are harder to debug (we can't know if the problem is within the model, the input or the ouput variables)

Reply:
Ok!